### PR TITLE
Refactor unit test driver as basis for feature test driver

### PR DESCRIPTION
**Pull request by johnjbarton**
TraceurTestRunner extends Mocha6 and contains common code for
unit and feature tests.  unitTestRunner is the instance for
unit tests.  The module that runs the unit tests is now
runUnitTests.js.

Rename testRunner to unitTestRunner.

### CODE REVIEW

File: test/modular/TraceurTestRunner.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* @fileoverview Configure mocha for Traceur testing. */
+
+import {Mocha6} from './Mocha6.js';
+let commander = require('commander');
```
Comment (arv) [inline test/modular/TraceurTestRunner.js]: Or 

let {Commander} = require...

File: test/modular/TraceurTestRunner.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    return this.defaultOptions_;
+  }
+
+  // Reads process arguments, merges defaults options, returns options object.
+  parseCommandLine() {
+    let testOptions = this.defaultOptions();
+    let commandLine =
+        new commander.Command(process.argv[0] + ' ' + process.argv[1]);
+
+    Object.getOwnPropertyNames(testOptions).forEach( (prop) => {
```
Comment (arv) [inline test/modular/TraceurTestRunner.js]: no ws between ( and (

File: test/modular/TraceurTestRunner.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      commandLine[prop] = testOptions[prop];
+    });
+
+    commandLine.option('-?, --help', 'Show this help text', () =>{
```
Comment (arv) [inline test/modular/TraceurTestRunner.js]: ws after `=>`

File: test/modular/TraceurTestRunner.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        process.exit(failed);
+      });
+    });
+  }
+
+  parseOptionsAndRun() {
+    this.applyOptions(this.parseCommandLine());
+    this.run();
+  }
+};
```
Comment (arv) [inline test/modular/TraceurTestRunner.js]: missing new line at EOF

File: test/unit/unitTestRunner.js [hunk 5]
```diff
@@ -12,21 +12,23 @@
...
-
-export var testRunner = new Mocha6({
-  ui: 'tdd',
-  ignoreLeaks: true,
-  importMetadata: {
-    traceurOptions: {
-      sourceMaps: 'memory'
-    }
-  }
+export var unitTestRunner = new TraceurTestRunner({
```
Comment (arv) [inline test/unit/unitTestRunner.js]: Wrong indentation/formattting:

```js
export var unitTestRunner = new TraceurTestRunner({
  patterns: [
    'test/unit/util/*.js',
    ...
  ]
});
```

**Comment by arv** [comment]
LGTM with a bunch of formatting nits.

File: test/modular/TraceurTestRunner.js (hunk 1 above)
Comment (johnjbarton) [inline test/modular/TraceurTestRunner.js]: done

File: test/modular/TraceurTestRunner.js (hunk 2 above)
Comment (johnjbarton) [inline test/modular/TraceurTestRunner.js]: done

File: test/modular/TraceurTestRunner.js (hunk 3 above)
Comment (johnjbarton) [inline test/modular/TraceurTestRunner.js]: done

File: test/modular/TraceurTestRunner.js (hunk 4 above)
Comment (johnjbarton) [inline test/modular/TraceurTestRunner.js]: done

File: test/unit/unitTestRunner.js (hunk 5 above)
Comment (johnjbarton) [inline test/unit/unitTestRunner.js]: done

**Pull request closed without merging**